src/core/utils.js

```diff
@@ -1,3 +1,113 @@
 'use strict'
 
+const CID = require('cids')
+const multihashes = require('multihashes')
+const promisify = require('promisify-es6')
+const map = require('async/map')
+
 exports.OFFLINE_ERROR = 'This command must be run in online mode. Try running \'ipfs daemon\' first.'
+
+/**
+ * Break an ipfs-path down into it's hash hash and an array of links.
+ *
+ * examples:
+ *  b58Hash -> { hash: 'b58Hash', links: [] }
+ *  b58Hash/mercury/venus -> { hash: 'b58Hash', links: ['mercury', 'venus']}
+ *  /ipfs/b58Hash/links/by/name -> { hash: 'b58Hash', links: ['links', 'by', 'name'] }
+ *
+ * @param  {String} ipfsPath An ipfs-path
+ * @return {Object}            { hash: base58 string, links: [string], ?err: Error }
+ * @throws on an invalid @param ipfsPath
+ */
+function parseIpfsPath (ipfsPath) {
+  const invalidPathErr = new Error('invalid ipfs ref path')
+
+  const matched = ipfsPath.match(/^(?:\/ipfs\/)?([^/]+(?:\/[^/]+)*)\/?$/)
+  if (!matched) {
+    throw invalidPathErr
+  }
```

Comment:
Can we please use the `is-ipfs` module for this?

Reply:
I really wanted to as well, and did with an earlier version but reverted it when I tried to enable custom hash algorithms as the `is-ipfs.isMultihash` helper does not work for non sha2-256 hashes.

I just looked again at the library and noticed `isCID` should cover our bases here; I've implemented it and added a test for those non default hashes for when we change it later.